main.go

```diff
@@ -36,15 +36,43 @@ func connectionChecker(peer smtpd.Peer) error {
 	return smtpd.Error{Code: 421, Message: "Denied"}
 }
 
+func addrAllowed(addr string, allowedAddrs []string) bool {
+	if allowedAddrs == nil {
+		return true
+	}
+
+	domidx := strings.LastIndex(addr, "@")
+	if domidx == -1 {
+		return false
+	}
+	domain := strings.ToLower(addr[domidx+1:])
+
+	for _, allowedAddr := range allowedAddrs {
+		allowedAddr = strings.ToLower(allowedAddr)
+
+		if strings.Index(allowedAddr, "@") == -1 {
```

Comment:
This is now a conflict because the absence of @ can be because it's a domain or a local address without domain yet.

Reply:
Per [below](https://github.com/decke/smtprelay/pull/9#issuecomment-778836237), I was unaware that "local" addresses could reach smtprelay. Updated (see below).